### PR TITLE
adding new module in HARFANGLAB -> Retrived agent hostname with their IP

### PR DESCRIPTION
**Pull request by lheronD**
adding new module in harfanglab-> Retrived agent hostname with their IP.

It returns all agents with this IP address, along with their name, IP, last visit date, OS and status.

There's also a boolean button to select only the last agent in the list.

### CODE REVIEW

**Comment by soc-dstny** [comment]
Hello, are you missing any information for the test workflow to be launched?

**Review by soc-dstny — APPROVED** [review]

File: HarfangLab/harfanglab/models.py [hunk 1]
```diff
@@ -29,3 +29,15 @@ class JobTriggerResult(BaseModel):
     action: str
     creationtime: str
     parameters: Optional[Dict[str, Any]]
+
+
+class HostnameEntry(BaseModel):
+    hostname: str
+    ipaddress: str
+    last_seen: str
+    os: str
```
Comment (squioc) [inline HarfangLab/harfanglab/models.py:38]: According to the Harfanglab documentation,
the name of the date when the agent was last seen is `lastseen`.
With the property `os`, do you mean `ostype`?

First suggestion: Rename the properties (don't forget to rename the usage also)
```suggestion
    lastseen: str
    ostype: str
```

Second suggestion: Use Field's alias
```suggestion
    last_seen: str = Field(..., alias='lastseen')
    os: str = Field(..., alias='ostype'
```

File: HarfangLab/harfanglab/get_hostnames_by_ip_action.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+        job_url = f"{instance_url}/api/data/endpoint/Agent/"
+        params: dict = {"ipaddress": target_ip}
+
+        response = requests.get(url=job_url, json=params, headers={"Authorization": f"Token {api_token}"})
+        response.raise_for_status()
+
+        data = response.json()
+
+        hostnames = [HostnameEntry(**entry) for entry in data]
```
Comment (squioc) [inline HarfangLab/harfanglab/get_hostnames_by_ip_action.py:31]: The agents are returned to the `results` property from the response
```json
{
    "count": 2,
    "results": [...]
}
```

We suggest the following change
```suggestion
        hostnames = [HostnameEntry(**entry) for entry in data.get("results", [])]
```

File: HarfangLab/action_harfanglab_gethostnamesbyip.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  },
+  "description": "Get the hostname of a machine by its IP address",
+  "docker_parameters": "harfanglab_get_hostnames_by_ip",
+  "name": "Hostname by IP",
+  "results": {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "properties": {
+      "hostnames": {
+        "description": "Hostnames",
+        "type": "object"
```
Comment (squioc) [inline HarfangLab/action_harfanglab_gethostnamesbyip.json:26]: According to the code, `hostnames` is a list of object

```suggestion
        "description": "Hostnames",
        "type": "array",
        "items": {
            "type": "object"
        }
```

**Review by squioc — CHANGES_REQUESTED** [review]
Hello.

Sorry for the delay.
We detected some issues in the action.
Can you fix them? Thank you

**Comment by soc-dstny** [comment]
I think the IC has gone through but the last check doesn't want to launch.

File: HarfangLab/harfanglab/models.py [hunk 4]
```diff
@@ -29,3 +29,15 @@ class JobTriggerResult(BaseModel):
     action: str
     creationtime: str
     parameters: Optional[Dict[str, Any]]
+
+
+class HostnameEntry(BaseModel):
+    hostname: str
+    ipaddress: str
+    lastseen: str = Field(..., alias="lastseen")
+    ostype: str = Field(..., alias="ostype")
```
Comment (squioc) [inline HarfangLab/harfanglab/models.py:38]: My bad. My previous comment was ambiguous.
As you renamed the fields like they are in the API response, you don't need to define an alias (Alias are useful when the name of the field in the model doesn't match the name of the field in the serialized structure)
I suggest removing the second part of the lines:
```suggestion
    lastseen: str
    ostype: str
```

File: HarfangLab/harfanglab/get_hostnames_by_ip_action.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        job_url = f"{instance_url}/api/data/endpoint/Agent/"
+        params: dict = {"ipaddress": target_ip}
+
+        response = requests.get(url=job_url, json=params, headers={"Authorization": f"Token {api_token}"})
+        response.raise_for_status()
+
+        data = response.json()
+
+        # Check if data is a dictionary with a key "result" or directly a list
+        result = data.get("result") if isinstance(data, dict) else data
```
Comment (squioc) [inline HarfangLab/harfanglab/get_hostnames_by_ip_action.py:32]: A little typo on the name of the field that contains the items in the API response:
```suggestion
        result = data.get("results") if isinstance(data, dict) else data
```

**Review by squioc — CHANGES_REQUESTED** [review]
Thank you for the changes.
Just a little fix to apply (and a suggestion).

Before I forget, can you also increase the number version in the [manifest.json](https://github.com/SEKOIA-IO/automation-library/blob/main/HarfangLab/manifest.json) file and add a little entry in the [CHANGELOG.md](https://github.com/SEKOIA-IO/automation-library/blob/main/HarfangLab/CHANGELOG.md)? Thank you

**Review by squioc — APPROVED** [review]
LGTM. Thanks for the contribution

**Pull request merged**